.github/workflows/homebrew-e2e.yml

```diff
@@ -0,0 +1,153 @@
+
+          echo ""
+          echo "=== 🔗 Checking bin wrapper ==="
+          WRAPPER_PATH=$(brew --prefix)/bin/badgetizr
+          echo "Wrapper path: $WRAPPER_PATH"
+          cat "$WRAPPER_PATH"
+
+          echo ""
+          echo "=== ✅ Verifying UTILS_PATH is set ==="
+          if grep -q "UTILS_PATH" "$WRAPPER_PATH"; then
+            echo "✅ UTILS_PATH found in wrapper"
+            grep "UTILS_PATH" "$WRAPPER_PATH"
+          else
+            echo "❌ UTILS_PATH not found in wrapper!"
+            echo "This would cause the v1.5.5 bug regression!"
+            exit 1
+          fi
+
+      - name: Test badgetizr --version
+        run: |
+          echo "🧪 Testing: badgetizr --version"
+          badgetizr --version
+
+      - name: Test badgetizr --help
+        run: |
+          echo "🧪 Testing: badgetizr --help"
+          badgetizr --help | head -20
+
+      - name: Test badgetizr error handling
+        run: |
+          echo "🧪 Testing: badgetizr without --pr-id (should fail with exit code 1)"
+
+          set +e
+          OUTPUT=$(badgetizr 2>&1)
+          EXIT_CODE=$?
+          set -e
+
+          echo "Exit code: $EXIT_CODE"
+          echo "Output: $OUTPUT"
+
+          if [ $EXIT_CODE -ne 1 ]; then
+            echo "❌ Expected exit code 1, got $EXIT_CODE"
+            exit 1
+          fi
+
+          if ! echo "$OUTPUT" | grep -q "Error.*--pr-id"; then
+            echo "❌ Expected error message not found in output"
+            exit 1
+          fi
+
+          echo "✅ Error handling validated (exit code 1 + correct error message)"
+
+      - name: Test providers are accessible
+        run: |
+          echo "🧪 Testing: providers directory is accessible"
+          INSTALL_PATH=$(brew --prefix)/opt/badgetizr/libexec
+          if [ -d "$INSTALL_PATH/providers" ]; then
+            echo "✅ Providers directory found"
+            ls -la "$INSTALL_PATH/providers"
+          else
+            echo "❌ Providers directory not found!"
+            exit 1
+          fi
+
+      - name: Cleanup
+        if: always()
+        run: |
+          echo "🧹 Uninstalling badgetizr..."
+          brew uninstall badgetizr || true
+
+          echo "🧹 Removing temporary tap..."
+          brew untap badgetizr-test/badgetizr || true
+          rm -rf $(brew --repository)/Library/Taps/badgetizr-test || true
+
+  homebrew-e2e-summary:
+    name: Homebrew E2E Summary
+    runs-on: ubuntu-latest
+    needs: [test-homebrew-installation]
+    if: always()
+
+    steps:
+      - name: Check all macOS versions passed
+        run: |
+          echo "=== Homebrew E2E Test Results ==="
+          echo "Matrix job result: ${{ needs.test-homebrew-installation.result }}"
+
+          # With fail-fast: false, we need to check the overall result
+          # Possible values: success, failure, cancelled, skipped
+          RESULT="${{ needs.test-homebrew-installation.result }}"
+
+          if [ "$RESULT" = "success" ]; then
```

Comment:
An if testing the same variable 3 tines because doing an else should be a switch